### PR TITLE
Refactor middleware, tests

### PR DESCRIPTION
**Pull request by ariskemper**
### Author should do the followings, if applicable

- [x] Add tests
- [x] Run tests
- [x] `yarn denoify` to generate files for Deno

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @ariskemper 

Thanks. I want to merge this. So, please run the `yarn format:fix`  to format the code.

**Comment by yusukebe** [comment]
@ariskemper 

Thanks, but please `denoify` again!

File: package.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
     "dist"
   ],
   "scripts": {
-    "test": "tsc --noEmit && vitest --run",
```
Comment (yusukebe) [inline package.json:13]: @ariskemper 

Sorry, I didn't notice the change here.

`tsc --noEmit` is necessary to check the types in the test phase.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
...
+    "test": "vitest --run",
+    "test:watch": "vitest --watch",
```
Comment (yusukebe) [inline package.json:14]: This change is not bad, but we want to be careful about changing `package.json`, so we should not introduce it in this PR.

Comment (ariskemper) [reply package.json:14]: @yusukebe will create a separate PR for this change, but i know vitest, that's why i made this change :)

**Comment by yusukebe** [comment]
@ariskemper 

Thanks! I'll merge this now.

**Pull request merged**